pbsync/__main__.py

```diff
@@ -31,6 +37,69 @@
 
 default_config_name = "PBSync.xml"
 
+retry_strategy = Retry(
+    total=5,  # Maximum number of retries
+    status_forcelist=[429, 500, 502, 503, 504],  # Retry on these status codes
+    backoff_factor=1  # Wait 1 sec before retrying, then increase by 1 sec each retry
+)
+
+adapter = HTTPAdapter(max_retries=retry_strategy)
+http = requests.Session()
+http.mount("https://", adapter)
+
+def check_gh_cli():
+    try:
+        result = subprocess.run(["gh", "--version"], capture_output=True, text=True, check=True)
+        pblog.info(result.stdout.strip())
+        return True
+    except FileNotFoundError:
+        return False
+
+def install_gh_cli():

```

Comment:
GitHub CLI is expected to be in `.github/git/gh.exe`,  please see `pbgh.py`. Though, I suppose we need to do some work to fix support for GitHub CLI while using repos on GitLab, since it uses GitLab CLI automatically in that case. So maybe forcing GitHub CLI is good there.